docs: Improve readme with new sections and fixes (#151)
moov-io/imagecashletter: README.md

## Usage
The Image Cash Letter project implements an HTTP server and [Go library](https://pkg.go.dev/github.com/moov-io/imagecashletter) for creating and modifying ICL files. We also have some [examples](https://pkg.go.dev/github.com/moov-io/imagecashletter/examples) of the reader and writer.

### Docker

We publish a [public Docker image `moov/imagecashletter`](https://hub.docker.com/r/moov/imagecashletter/) from Docker Hub or use this repository. No configuration is required to serve on `:8083` and metrics at `:9093/metrics` in Prometheus format. We also have Docker images for [OpenShift](https://quay.io/repository/moov/imagecashletter?tab=tags) published as `quay.io/moov/imagecashletter`.

Pull & start the Docker image:
```
docker pull moov/imagecashletter:latest
docker run -p 8083:8083 -p 9093:9093 moov/imagecashletter:latest
```

List files stored in-memory:
```
curl localhost:8083/files
```
```
null
```

> Create a file on the HTTP server:

Reviewer: ```suggestion
Upload a x9 file (binary)
```

```
curl -XPOST --data-binary "@./test/testdata/valid-ascii.x937" http://localhost:8083/files/create
```
```
{"id":"<YOUR-UNIQUE-FILE-ID>","fileHeader":{"id":"","standardLevel":"03","testIndicator":"T","immediateDestination":"061000146","immediateOrigin":"026073150", ...
```

> Read the X9 file (in JSON form):

Reviewer: ```suggestion
Retrieve an existing x9 file (JSON)
```

```
curl http://localhost:8083/files/<YOUR-UNIQUE-FILE-ID>
```
```
{"id":"<YOUR-UNIQUE-FILE-ID>","fileHeader":{"id":"","standardLevel":"03","testIndicator":"T","immediateDestination":"061000146","immediateOrigin":"026073150", ...
```

> Create a file with JSON format on the HTTP server:

Reviewer: ```suggestion
Create an x9 file from JSON
```

```
curl -XPOST -H "content-type: application/json" localhost:8083/files/create --data @./test/testdata/icl-valid.json
```
```
{"id":"<YOUR-UNIQUE-FILE-ID>","fileHeader":{"id":"","standardLevel":"35","testIndicator":"T","immediateDestination":"231380104","immediateOrigin":"121042882", ...
```

Get the formatted file:
```
curl localhost:8083/files/<YOUR-UNIQUE-FILE-ID>/contents
```
```
P0135T231380104121042882201810032219NCitadel      Wells Fargo    US   P100123138010412104288220181003201810032219IGA1   Contact Name 5558675552  P200123138010412104288220181003201810039999   1  01             P25   123456789 031300012       555888100001000001       GD1Y030BP261121042882201810031       938383      01  Test Payee   Y10
...
```

> ### Google Cloud Run

Reviewer: @adamdecaf This section looks good to me, but you should probably double check since you're more familiar with GCP.

Reviewer 2: I don't know. We just had the Google Cloud Run button for ACH. Can we try this prior to merge?

Reviewer: I can try it out. I've got a GCP account.

Author: I was going to use the run button originally, but the port number (8083) can't be defined through the button URL, so I went for a longer workaround.

Reviewer: Looks like you can supply the port in configuration file: https://github.com/GoogleCloudPlatform/cloud-run-button#customizing-deployment-parameters 🤷‍♂️

Reviewer: I tested out the steps as written and they worked great. I had never deployed an app to GCP, so I'd say this was a good test.

Author: Ok awesome, thanks for going through them.

$ go doc github.com/moov-io/imagecashletter CashLetter
```

The package [`github.com/moov-io/imagecashletter`](https://pkg.go.dev/github.com/moov-io/imagecashletter) offers a Go-based Image Cash Letter file reader and writer. To get started, check out a specific example:

 ICL File | Read | Write |
|---------|------|-------|
| [Link](examples/imagecashletter-read/iclFile.x937) | [Link](examples/imagecashletter-read/main.go) | [Link](examples/imagecashletter-write/main.go) |

> There are a few reader and writer options available to change default behavior:

Reviewer: ```suggestion
ImageCashLetter's file handling behaviors can be modified to accommodate your specific use case. This is done by passing _options_ into ICL's `reader` and `writer` during instantiation. For example, to read EBCDID encoded files you would instantiate a reader with `NewReader(fd, ReadVariableLineLengthOption(), ReadEbcdicEncodingOption())`.
```
The following options are currently supported:

| Option | Description |
|-----|-----|

> | `ReadVariableLineLengthOption` | Allows Reader to split ICL files based on encoded line lengths. |

Reviewer: ```suggestion
| `ReadVariableLineLengthOption` | Allows Reader to split ICL files based on the Inserted Length Field. |
```

| `ReadEbcdicEncodingOption` | Allows Reader to decode scanned lines from EBCDIC to UTF-8. |

> | `WriteVariableLineLengthOption` | Allows Writer to write control bytes ahead of record to describe how long the line is. |

Reviewer: ```suggestion
| `WriteVariableLineLengthOption` | Instructs the Writer to begin each record with the appropriate Inserted Length Field. |
```
